### PR TITLE
Makes sand/snowdigging not rely on archaeology feature for the interaction

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Simply put, this component with archdrops is responsible for hundreds of thoudsands of total lists being made, along with around 150 thousand callback objects, 50+ thousand components, all so that you can get sand or snow if you click on an asteroid tile. This translates to 50-100 MB of memory depending on how deep space generation shakes out. Yes, really.

:cl: Naksu
del: archaeology component no longer handles sand-digging, it has been disabled until someone works on xenoarch again.
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
RIP

**Comment by ExcessiveUseOfCobblestone** [comment]
Archaeology is probably not even component-worthy, as open turfs could probably handle it entirely.

File: code/game/turfs/simulated/floor/plating/asteroid.dm [hunk 1]
```diff
@@ -73,9 +97,6 @@
 /turf/open/floor/plating/asteroid/basalt/Initialize()
 	. = ..()
 	set_basalt_light(src)
-	GET_COMPONENT(arch, /datum/component/archaeology)
-	ASSERT(isnull(arch.callback))
-	arch.callback = CALLBACK(src, /atom/proc/set_light, 0)
```
Comment (SpaceManiac) [inline code/game/turfs/simulated/floor/plating/asteroid.dm]: Is this call being maintained? It's used to make glowing turfs stop glowing when they're dug.

Comment (vuonojenmustaturska) [reply code/game/turfs/simulated/floor/plating/asteroid.dm]: it wasn't before, now is.

**Review by SpaceManiac — APPROVED** [review]
:pray:

**Pull request merged**